docs/proposals/standardize-ring-monitoring.md

```diff
@@ -0,0 +1,166 @@
+
+**Author:** @bubu11e
+
+**Date:** 11/2023
+
+**Sponsor(s):**
+
+**Type:** Feature
+
+**Status:** Draft
+
+**Related issues/PRs:**
+
+---
+
+## Background
+
+Mimir is using rings to ensure processes of sharding and replication. The following rings are used inside mimir:
+
+- distributor
+- ingester
+- compactors
+- store-gateway
+- ruler (optional)
+- alertmanager (optional)
+- query-scheduler (optional)
+
+Community based monitoring is currently ensured by three alerting rules:
+
+- [MimirIngesterUnhealthy](https://github.com/grafana/mimir/blob/main/operations/mimir-mixin-compiled-baremetal/alerts.yaml#L4)
+- [MimirAlertmanagerRingCheckFailing](https://github.com/grafana/mimir/blob/main/operations/mimir-mixin-compiled-baremetal/alerts.yaml#L534)
+- [MimirRulerFailedRingCheck](https://github.com/grafana/mimir/blob/main/operations/mimir-mixin-compiled-baremetal/alerts.yaml#L441)
+
+## Problem Statement
+
+The previsoulsy mentionned rules does not cover monitoring of all the rings available in Mimir.
+
+Furthermore their implementation differs:
+
+- MimirIngesterUnhealthy alerting rule is based on metric `cortex_ring_members`available for all rings in the cluster
+
+  - Probe does not take into account availability zones
+  - Currently only focused on ingester ring
+  - Does not indicate if the ring is broken or not but only if unhealthy instances have been found in it which could lead to unwanted alerts inside large cluster.
+
+- MimirAlertmanagerRingCheckFailing alerting rule is based on metric `cortex_alertmanager_ring_check_errors_total` only available for alertmanager
+
+  - Only apply to alertmanager ring
+  - Expose ring usability
+  - Does not alert about an unused broken ring.
+  - This method needs to be reimplemented for every ring
+
+- MimirRulerFailedRingCheck alerting rule is based on metric `cortex_ruler_ring_check_errors_total` only available for ruler
+  - Only apply to ruler ring
+  - Give an interesting point a view about the usability of the ring.
+  - Does not alert about an unused broken ring.
+  - This method needs to be reimplemented for every ring
+
+## Goals
+
+- Monitor all rings used inside Mimir software
+- Give both a view of the state of the ring and of its usability
+- Standardize monitoring of rings
+- Automatically adapt to presents rings are some of thems are optionals
+
+## Non-Goals (optional)
+
+- Improve monitoring of other dskit based tools(loki/tempo/pyroscope) even if it could be a nice addition.
+
+## Proposals
+
+### Proposal 0: Do nothing
+
+Current solution has proven to be working. The most critical ring (a.k.a ingester) is monitored.
+
+**Pros:**
+
+- Nothing to be done
+
+**Cons:**
+
+- Not all mimir rings are monitored
+- Not homogenous among all rings
+- Alert does not taking into account availability zone
+- No indication on the ring state, only if there are unhealthy nodes and only for the ingester ring
+
+### Proposal 1: Make alert MimirIngesterUnhealthy more generic
```

Comment:
some of the rings are configured with auto-forget. Like the ruler and store-gateway and compactor. That means that unhealthy members will be recovered from and that shouldn't lead to significant downtime for the system (there will be missed rule evals and delayed compactions for example)

because of this I think this would make this alert unactionable if it were enabled for all rings. I think the rings that don't have [auto-forget](https://github.com/grafana/mimir/blob/7fbf8af80cfc3c496268faba45d3c5cbb86c75d6/vendor/github.com/grafana/dskit/ring/basic_lifecycler_delegates.go#L122) would make sense to share this alert.